plugin-java/src/main/java/appland/execution/AppMapJavaConfigUtil.java

```diff
@@ -50,24 +33,18 @@ private AppMapJavaConfigUtil() {
                                                                           @NotNull VirtualFile context) {
         ApplicationManager.getApplication().assertReadAccessAllowed();
 
-        // content root of the module, which contains the context file
-        var matchingModuleRoot = Arrays.stream(ModuleRootManager.getInstance(module).getContentRoots())
-                .filter(root -> VfsUtilCore.isAncestor(root, context, false))
-                .findFirst()
-                .orElse(null);
-        if (matchingModuleRoot != null) {
-            return matchingModuleRoot;
-        }
-
-        // Some project models, e.g. Gradle, create multiple modules for a single gradle project.
-        // For example, Gradle module "main" with content root "<root>/module-one/src/main"
-        // and a top-level module with content root "<root>/module-one".
-        // The context (e.g. "<root>/module-one") may be located outside the root of module (e.g "main").
-        var projectContentRoot = ProjectRootManager.getInstance(module.getProject())
-                .getFileIndex()
-                .getContentRootForFile(context, false);
-        if (projectContentRoot != null) {
-            return projectContentRoot;
+        // Any of the module's roots will do
+        var moduleRoot = ModuleRootManager.getInstance(module).getContentRoots()[0];
+        var projectRoots = ProjectRootManager.getInstance(module.getProject()).getContentRoots();
```

Comment:
IMHO it would be better to look at all module content roots (even though most will contain 1 root only).
There's a `min` method on streams to help choosing the shortest path.

Perhaps something like the following?
```java
        var moduleRoots = ModuleRootManager.getInstance(module).getContentRoots();
        var projectRoots = ProjectRootManager.getInstance(module.getProject()).getContentRoots();
        var candidateProjectRoot = Arrays.stream(projectRoots)
                .filter(contentRoot -> Arrays.stream(moduleRoots).anyMatch(moduleRoot -> {
                    return VfsUtilCore.isAncestor(contentRoot, moduleRoot, false);
                }))
                .min(Comparator.comparingInt(file -> file.getPath().length()));

        if (candidateProjectRoot.isPresent()) {
            // The candidate roots are sorted, so  the top-level one will be
            // first (because it will be shortest).
            return candidateProjectRoot.get();
        }
```

But I'm currently not sure, what would happen if we returned a project content root, which does not contain the context file. Have you seen something like this or what do you think about this edge-case?